tensorflow/compiler/mlir/lite/transforms/optimize.cc

```diff
@@ -197,9 +197,9 @@ TypeAttr RescaleQtype(Type input, Attribute factor) {
 DenseElementsAttr GetShape(Value output_val) {
   auto output_type = output_val.getType().cast<RankedTensorType>();
   auto shape_vector = output_type.getShape();
-  std::vector<int32_t> shape(shape_vector.size());
-  for (int i = 0; i < shape_vector.size(); ++i) {
-    shape[i] = shape_vector[i];
+  std::vector<int32_t> shape;
+  for (auto shape_object : shape_vector) {
+    shape.push_back(shape_object);
```

Comment:
This changes memory allocation behavior as now the vector keeps being regrown. How about this?
```suggestion
  std::vector<int32_t> shape;
  shape.reserve(shape_vector.size());
  for (auto shape_object : shape_vector) {
    shape.push_back(shape_object);
```